File: java/user-management-policy/src/main/java/com/hexagonalchess/usermanagement/FriendshipRepository.java
package com.hexagonalchess.usermanagement;

import java.util.List;

public interface FriendshipRepository {
    void save(Friendship friendship);
    List<Friendship> findByOwnerId(UserId ownerId);
    void deleteByOwnerAndFriend(UserId ownerId, UserId friendId);
}

File: ruby/user_management_policy/lib/user_management/friendship_repository.rb
module UserManagement
  module FriendshipRepository
    def save(friendship) = raise NotImplementedError
    def find_by_owner_id(owner_id) = raise NotImplementedError
    def delete_by_owner_and_friend(owner_id, friend_id) = raise NotImplementedError
  end
end
